pkg/readiness/ready_tracker_test.go

```diff
@@ -97,7 +97,7 @@ func setupOpa(t *testing.T) *opa.Client {
 }
 
 func setupController(mgr manager.Manager, wm *watch.Manager, opa *opa.Client, mutationSystem *mutation.System) error {
-	tracker, err := readiness.SetupTracker(mgr, mutationSystem != nil)
+	tracker, err := readiness.SetupTracker(mgr, mutationSystem != nil, false)
```

Comment:
Should we update this function to check for providerCache dynamically and then pass in `nil` in the tests so we can add tests for externaldata later.

Reply:
added test